services/authService.ts

```diff
@@ -76,9 +76,49 @@ const loginOrSignupWithGoogle = async (
   }
 }
 
+/**
+ * Login or signUp with Microsoft
+ * @param microsoftProfile{Object} : Microsoft profile response from Microsoft OAuth2.0
+ */
+ const loginOrSignupWithMicrosoft = async (
+  microsoftProfile: MicrosoftOAuthJson
+): Promise<Users> => {
+  try {
+    const user = await prisma.users.findUnique({
+      where: {
+        email: microsoftProfile.mail ?? microsoftProfile?.userPrincipalName
```

Comment:
What is the value in `microsoftProfile?.userPrincipalName`? If its a name, how can we use it in place of an email?

Reply:
https://learn.microsoft.com/en-us/azure/active-directory/hybrid/plan-connect-userprincipalname#what-is-userprincipalname


![Screenshot from 2022-11-13 17-51-47](https://user-images.githubusercontent.com/34452139/201521416-84a1cca6-d64d-484f-8d7e-fb6b7d7fe12a.png)
https://learn.microsoft.com/en-us/graph/api/resources/user?view=graph-rest-1.0